pkg/provider/aws/resources/vpc.go

```diff
@@ -363,6 +363,28 @@ func NewSubnet(subnetName string, vpc *Vpc, cidrBlock string, subnetType string,
 	}
 }
 
+// IsPublic returns whether this Subnet is public within the resource graph. A subnet is public iff its upstream
+// RouteTable has a downstream InternetGateway.
+func (subnet *Subnet) IsPublic(dag *core.ResourceGraph) bool {
+	for _, upstreamFromSubnet := range dag.GetUpstreamResources(subnet) {
+		routeTable, ok := upstreamFromSubnet.(*RouteTable)
+		if !ok {
+			continue
+		}
+		for _, downstreamFromRouteTable := range dag.GetAllDownstreamResources(routeTable) {
+			if _, isIGW := downstreamFromRouteTable.(*InternetGateway); isIGW {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
big fan of doing it this way

Reply:
Will do in a separate PR, if that's okay. I have a TODO on my personal list to get rid of the `.Type` field altogether and just use this func, so I can add a test then.

Reply:
Unfortunately, `.Type` needs to stay for VPC imports since those don't set up InternetGateways.